vim-engine/src/main/kotlin/com/maddyhome/idea/vim/common/Graphemes.kt

```diff
@@ -0,0 +1,255 @@
+package com.maddyhome.idea.vim.common
+
+/**
+ * Move over unicode extended grapheme cluster boundaries.
+ *
+ * https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries
+ */
+public object Graphemes {
+  public fun next(charSeq: CharSequence, start: Int): Int? {
+    if (start < 0 || start >= charSeq.length) return null
```

Comment:
I think it's better to throw an error in this case

Reply:
The second half of the condition (`start >= charSeq.length`) is like that, so we can call `next` in a loop and have `null` as a break condition instead of tracking the offset. `start < 0` makes zero sense here though, and for sure should throw. Would you prefer to make the return type non-nullable and move the check up to the caller's responsibility?

Reply:
I see, thank you for the explanation. I think we can add this contract to the KDoc and keep it like this.